include/flamegpu/runtime/messaging/MessageSpatial2D/MessageSpatial2DDevice.cuh

```diff
@@ -500,6 +500,13 @@ class MessageSpatial2D::In {
              // Return iterator at min corner of env, this should be safe
              return WrapFilter(metadata, metadata->min[0], metadata->min[1]);
          }
+         if (fmodf(metadata->max[0] - metadata->min[0], metadata->radius) > 0.00001f ||
```

Comment:
Any particular reason/justification for this particular epsilon value?.

Though our current spatial comms is fp32 only anyway, so a magic value is fine, just why this one / is it sufficient for different values

E.g. if the comm rad is a small positive number, will this work? (though if its too small then large env widths will cause other problems anyway, so can prolly get away with a fixed arbitrary epsilon)

Tests only appear to cover a single radii/width combo.

Reply:
(I'd rather make wrapped work at a tiny cost to perf for non-wrapped to avoid this floating point crap)

Reply:
I agree that hurting wrapped more and not hurting non-wrapped is the better choice. 

Given the radii and width are known on the host, we could maybe do the check there once, and then set a value on the device somewhere marking if wrapped is possioble for that message list or not? for less perf loss to wrapped usage. An extra global memory read is probably cheaper than a couple of fmodf's?  

Could do it entirley on the host if we made wrapped opt-in on the message list, but I assume that's not desirable? 

Not sure there's a better option than fmodf, jsut the particular epsilon might need some thought / broader testing.